src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2733,10 +2735,100 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
     }
 
+    winrt::fire_and_forget TermControl::_updateSelectionMarkers(IInspectable /*sender*/, Control::UpdateSelectionMarkersEventArgs args)
+    {
+        auto weakThis{ get_weak() };
+        co_await resume_foreground(Dispatcher());
+        if (weakThis.get() && args)
+        {
+            if (_core.HasSelection() && !args.ClearMarkers())

```

Comment:
cases where there is a selection, but `ClearMarkers==true`:
* Mouse selection, via `SetEndSelectionPoint`
* `LeftClickOnTerminal`, pretty self-explanatory
* A selection created from searching for text

okay those all make sense

Reply:
copied this into the PR body.